### DISCUSSION TITLE
Related to switch user

### DISCUSSION THREAD

**Discussion by singhragvendra503** [containers/podman#22381] [Q&A] [answered]
I have created a user `devops` in Ubuntu 22.04 using that I can run podman containers. There are two users `devops` and `ubuntu`, `ubuntu` has super user permission. `devops` user can't add in sudoers file. 
My question is, when I am switch user from `ubuntu` to `devops` using 
```
sudo su devops
```
I can run containers but if I switch user from `ubuntu` to `devops` using
```
su devops
```
I got a `ERROR`
`ERRO[0000] XDG_RUNTIME_DIR directory "/run/user/1000" is not owned by the current user `

**Comment by eriksjolund** [comment]
Instead of using `su` it's better to use `machinectl` or `systemd-run`

#### Run podman as another user interactively

```
$ sudo -i
# machinectl shell --uid=username
Connected to the local host. Press ^] three times within 1s to exit session.
$ podman run --rm -ti docker.io/library/alpine
/ # cat /etc/alpine-release 
3.19.1
/ # 
```

#### Run podman as another user noninteractively 

```
# sudo systemd-run \
         --machine=username@ \
         --quiet \
         --user \
         --collect \
         --pipe \
         --wait \
           podman run \
              --rm \
              docker.io/library/alpine \
                  echo hello
hello
```


As to why not use `su` see the blog post

 _Why can't I run rootless Podman containers when I log into a user via sudo or su?_ https://www.redhat.com/sysadmin/sudo-rootless-podman

See also the tip in troubleshooting.md
[Podman run fails with ERRO[0000] XDG_RUNTIME_DIR directory "/run/user/0" is not owned by the current user or Error: creating tmpdir: mkdir /run/user/1000: permission denied.](https://github.com/containers/podman/blob/main/troubleshooting.md#31-podman-run-fails-with-erro0000-xdg_runtime_dir-directory-runuser0-is-not-owned-by-the-current-user-or-error-creating-tmpdir-mkdir-runuser1000-permission-denied)

**Comment by singhragvendra503** [comment]
I am logging in using `sudo machinectl shell --uid=1002` and running the container. Whenever logged out the container stopped.

**Reply by eriksjolund** [reply, marked as answer]
Probably it will work if you enable linger
```
sudo loginctl enable-linger 1002
```
See tip in troubleshooting.md 
[rootless containers exit once the user session exits](https://github.com/containers/podman/blob/main/troubleshooting.md#17-rootless-containers-exit-once-the-user-session-exits)
